src/mlpack/tests/rl_environment_test.cpp

```diff
@@ -0,0 +1,46 @@
+/**
+ * @file rl_environment_test.hpp
+ * @author Shangtong Zhang
+ *
+ * Basic test for the reinforcement learning task environment
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/core.hpp>
+
+#include <mlpack/methods/reinforcement_learning/environment/mountain_car.hpp>
+#include <mlpack/methods/reinforcement_learning/environment/cart_pole.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::rl;
+
+BOOST_AUTO_TEST_SUITE(RLEnvironmentTest)
+
+BOOST_AUTO_TEST_CASE(MountainCarTest)
+{
+  const auto task = MountainCar();
```

Comment:
I'd like to avoid the use of auto here because I think in this case it doesn't enhance code readability; in fact, a user could take a look at the test case to get an impression on how to use the code and might wondering what action, reward, etc. is. Of course, he could look into the code or get it from the context but since the type isn't complex we could just use the type instead of auto here. Let me know what you think.